src/kakarot/precompiles/datacopy.cairo

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.cairo.common.memcpy import memcpy
+
+// Internal dependencies
+from kakarot.model import model
+from utils.utils import Helpers
+from kakarot.memory import Memory
+from kakarot.execution_context import ExecutionContext
+
+// @title DataCopy precompile
+// @custom:precompile
+// @custom:address 0x04
+// @notice This precompile serves as a cheaper way to copy data in memory
+// @author @abdelhamidbakhta
+// @custom:namespace PrecompileDataCopy
+namespace PrecompileDataCopy {
```

Comment:
is the name "identity" or "data copy"? https://www.evm.codes/precompiled#0x04?fork=grayGlacier